src/endpoint/s3/ops/s3_get_bucket.js

```diff
@@ -30,7 +30,18 @@ async function get_bucket(req) {
             (cont_tok_to_key_marker(cont_tok) || start_after) : req.query.marker,
     };
 
+    if (req.query.get_from_cache) {
+        params.get_from_cache = req.query.get_from_cache;
+    }
     const reply = await req.object_sdk.list_objects(params);
+
+    // Update the object creation time in case this is from cache
+    if (req.query.get_from_cache) {
```

Comment:
I don't think this is the condition is needed, I think obj.last_modified_time should always have precedence over create_time if the former is set.